keyboards/treadstone32/keymaps/default/keymap.c

```diff
@@ -0,0 +1,188 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+extern keymap_config_t keymap_config;
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layer_number {
+  _BASE = 0,
+  _LOWER,
+  _RAISE,
+  _ADJUST,
+};
+
+enum custom_keycodes {
+  RGBRST = SAFE_RANGE,
+  LOWER,
+  RAISE,
+  KANJI,
+};
+
+// enum tapdances{
+//   TD_CODO = 0,
+//   TD_SLRO,
+// };
+
+// Layer Mode aliases
+#define KC_MLAD  MO(_ADJUST)
+
+#define KC______ KC_TRNS
+#define KC_XXXXX KC_NO
+#define KC_KANJI KANJI
+
+// Adjust layer keys
+#define KC_RST   RESET
+#define KC_LRST  RGBRST
+#define KC_LTOG  RGB_TOG
+#define KC_LHUI  RGB_HUI
+#define KC_LHUD  RGB_HUD
+#define KC_LSAI  RGB_SAI
+#define KC_LSAD  RGB_SAD
+#define KC_LVAI  RGB_VAI
+#define KC_LVAD  RGB_VAD
+#define KC_LMOD  RGB_MOD
+#define KC_KNRM  AG_NORM
+#define KC_KSWP  AG_SWAP
+
+// Base layer mod tap
+#define KC_A_SF  LSFT_T(KC_A)
+#define KC_Z_CT  LCTL_T(KC_Z)
+#define KC_X_AL  LALT_T(KC_X)
+#define KC_C_GU  LGUI_T(KC_C)
+#define KC_SSCT  LCTL_T(KC_SLSH)
+#define KC_ENSF  LSFT_T(KC_ENT)
+
+// Lower layer mod tap
+#define KC_F6SF  LSFT_T(KC_F6)
+#define KC_BSSF  LSFT_T(KC_BSLS)
+#define KC_11CT  LCTL_T(KC_F11)
+#define KC_12AL  LALT_T(KC_F12)
+
+// Layer tap
+#define KC_BSLO  LT(_LOWER, KC_BSPC)
+#define KC_SPRA  LT(_RAISE, KC_SPC)
+
+// Tap dance
+// #define KC_CODO  TD(TD_CODO)
+// #define KC_SLRO  TD(TD_SLRO)
+
+// qk_tap_dance_action_t tap_dance_actions[] = {
+//   [TD_CODO] = ACTION_TAP_DANCE_DOUBLE(KC_COMM, KC_DOT),
+//   [TD_SLRO] = ACTION_TAP_DANCE_DOUBLE(KC_SLSH, KC_RO),
+// };
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT( \
```

Comment:
The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater and less "hack-y".

Reply:
I felt that I could not delete it before, but I was able to delete it. Thank you.